### ISSUE TITLE
GpgError: exit code 2

### ISSUE THREAD

**Issue by sciurius** [keybase/keybase-issues#821]
(Referring to issue #817 -- I removed the additional keyring from my gnupg config)

% keybase track someonelse
warn: When checking sciurius: GpgError: exit code 2
warn: Likely this is a bug or transient error; but the server could be compromised
✔ public key fingerprint: 60B5 6330 872C FD47 B337 0B34 0E62 6BAD 1298 C2B4
✔ "sciurius" on github: https://gist.github.com/78b3d920d008336ba127
✔ admin of www.squirrel.nl via HTTP: http://www.squirrel.nl/keybase.txt
Is this you? [y/N] 

Is this a bug, or transient error, or is the server compromised?

**Comment by seefood** [comment]
Yup, seems like the same thing is happening here. My keyring goes back to 1994. I did an --update-trustdb and was amused to be asked about keys of friends from Fidonet (BBS network, pre commercial internet and Emails) with keys of 1023R, 768R and even 512R and 384R. We were young and naiive with out 386SXs...

So, should I try to export all the private keys, kill my keyrings entirely and re-import?
I tried this: http://stuff-things.net/2015/04/22/gpg-public-key-of-ultimately-trusted-key-00000000-not-found/

I had 21 keys ending with 00000000, removed them, re-imported, checked and I get the error again, I export again and see two came back.

on the second iteration I import with gpg2, check with gpg2 goes fine, then a check with gpg pops up the error again, and then it's there for gpg2 as well. ARRGH!

bottom line, gpg 1.4.18 (Ubuntu) breaks the trustdb, even if I only run a check (shouldn't that be a RO operation?), however I can't remove it because apt depends on it (and not gpg2). this is annoying as hell :-(

anyway, I'm glad keybase defaults to using gpg2, and I managed to login. Thanks @maxtaco!

**Comment by a11enh** [comment]
Seems like I'm in a similar boat to Seefood - I too was using gpg back in the early days!  I also am getting a 00000 error... I think it's related to old md5 keys in the keychain.  Were you able to solve the problem @seefood  ?  To make matters worse, I went ahead and bought a hardware key and the master lives there for my current keypair.  LOL- so not sure how easy it will be to rebuild the keychain from scratch.

**Comment by CarloWood** [comment]
Hi, I had ownertrust keys that ended in 00000000 too (and removed them like a11enh did as per seefood's link).

The REASON I got these keys, imho, is because I JUST upgraded from gpg 1.4.22 to gpg 2.2.4, which silently deleted (skipped) two important secret keys that I was still using (an RSA 1024 from 1997).

My guess is that I still had trust signatures (by those keys, or others that old) in my trust db and they are simply screwing things up during the transition from 1.4 to 2.2 because 2.2 can't understand them anymore.

Seems a real bug in gpg to me therefore.
